### ISSUE TITLE
OS-level *.ipynb open support

### ISSUE THREAD

**Issue by flutefreak7** [jupyterlab/jupyterlab#7619]
tldr: Notebooks should be double-click-able / executable the way other files are for users who run jupyterlab locally.

Now that JupyterLab supports opening notebooks based on a URL and supports having multiple workspaces managed with URL's it seems like it should be possible to make a file-handler at the OS level that converts a filepath to a URL and opens it in the locally running jupyter server instance.

I completely understand that Jupyter is designed with client-server separation as a core tenant, but the user experience as a desktop application is pretty frustrating. If a desktop file handler could be developed that would open a notebook in a the current local jupyterlab session (or possibly start a jupyterlab session if needed), that would make it much easier to use notebooks with the rest of the OS. Notebooks could be opened directly from a filebrowser, from an email, or from the command-line. Ideally, we should be able to interact with notebooks the same way we interact with other files, like an Excel Workbook for example.

**Comment by vidartf** [comment]
Ref prior art: [nbopen](https://github.com/takluyver/nbopen)

**Comment by vidartf** [comment]
Also relevant: https://github.com/jupyterlab/jupyterlab_app

**Comment by vidartf** [comment]
If I considered it solved I would close the issue 😉. I was merely cross-referencing other work related to the issue, so that others who look at this issue might have a better context. 

If you know of a solution that works, please consider helping out by creating a PR with the proposed solution 😊

**Comment by flutefreak7** [comment]
I don't have the bandwidth to steward the issue, but I'll ask the first question that occurs to me.

Are these potential solutions mature? Even if not, is there consensus in the community that these efforts are the primary effort to solve this concern? If so then, yeah this issue should be considered closed and point to those solutions either as the "solution" or at least the best path towards solution.

**Comment by frenzymadness** [comment]
My plan is to open a PR proposing a .desktop file which makes it much easier to use lab without opening a terminal window first. It's an improvement but it probably does not solve all the issues.
